caikit_health_probe/__main__.py

```diff
@@ -205,8 +201,17 @@ def _grpc_health_probe(
         )
         import grpc  # pylint: disable=import-outside-toplevel
 
+    # Server hostname to use unless using socket mode
     hostname = f"localhost:{port}"
-    if tls_key and tls_cert:
+    socket_file = get_config().runtime.grpc.unix_socket_path
+
+    # If available, use a unix socket
+    if os.path.exists(os.path.dirname(socket_file)):
```

Comment:
Oops, this is not safe

```suggestion
    if socket_file and os.path.exists(os.path.dirname(socket_file)):
```

Reply:
oh good catch